lib/std/zig/string_literal.zig

```diff
@@ -340,6 +340,7 @@ pub fn parseWrite(writer: anytype, bytes: []const u8) error{OutOfMemory}!Result
                         if (bytes[escape_char_index] == 'u') {
                             var buf: [4]u8 = undefined;
                             const len = utf8Encode(codepoint, &buf) catch {
+			        // Note: never fails as codepoint is in range and surrogates are allowed, so this catch clause cannot be reached.
                                 return Result{ .failure = .{ .invalid_unicode_codepoint = escape_char_index + 1 } };
                             };
```

Comment:
If an error is unreachable then it should be made explicit with `unreachable`:
```suggestion
                            const len = utf8Encode(codepoint, &buf) catch |err| switch (err) {
                                error.CodepointTooLarge => unreachable, // Checked in `parseEscapeSequence`.
                            };
```

I think it'd also be good to explicitly use `utf8EncodeAllowSurrogates` without shortening it.

Reply:
Great feedback, thanks!  😊  Will update the code as suggested.